core/src/mindustry/input/InputHandler.java

```diff
@@ -1161,7 +1162,17 @@ void iterateLine(int startX, int startY, int endX, int endY, Cons<PlaceLine> con
         }
 
         if(diagonal){
-            points = Placement.pathfindLine(block != null && block.conveyorPlacement, startX, startY, endX, endY);
+            if(block != null && block instanceof Conveyor){
+                Tile start = world.tile(startX, startY);
+                Tile end = world.tile(endX, endY);
+                if(start.block() instanceof Conveyor && end.block() instanceof Conveyor){
+                    points = Placement.upgradeConveyorLine(startX, startY, endX, endY);
+                }else{
+                    points = Placement.pathfindLine(true, startX, startY, endX, endY);
+                }
```

Comment:
these don't seem to be used anywhere other than the if right below it`:

```suggestion
                if(world.tile(startX, startY).block() instanceof Conveyor && world.tile(endX, endY).block() instanceof Conveyor){
                    points = Placement.upgradeConveyorLine(startX, startY, endX, endY);
                }else{
                    points = Placement.pathfindLine(true, startX, startY, endX, endY);
                }
```